src/Chapter4.hs

```diff
@@ -488,7 +494,36 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
-
+mapList :: (a -> b) -> List a -> List b
+mapList _ Empty = Empty
+mapList fun (Cons x tl) = Cons (fun x) (mapList fun tl)
+
+reverseList :: List a -> List a
+reverseList lst = go lst Empty
+  where
+    go :: List a -> List a -> List a
+    go Empty acc = acc
+    go (Cons hd tl) acc = go tl (Cons hd acc)
+
+concatList :: List a -> List a -> List a
+concatList lst1 lst2 = go (reverseList lst1) lst2
```

Comment:
This is quite a costy implementation. Any reasons to use `reverse` in here?

Also, it is better to name the function not `concat` as what it actually does is appending 🙂 `concat` in Haskell has a bit another meaning of inlining (`[[a]] -> [a]`).